src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -319,10 +333,17 @@ private IOperation CreateBoundLocalOperation(BoundLocal boundLocal)
             ITypeSymbol type = boundLocal.Type;
             Optional<object> constantValue = ConvertToOptional(boundLocal.ConstantValue);
             bool isImplicit = boundLocal.WasCompilerGenerated;
+            if (isDeclaration && syntax is DeclarationExpressionSyntax declarationExpressionSyntax)
+            {
+                syntax = declarationExpressionSyntax.Designation;
+                Lazy<IOperation> localReferenceExpression = new Lazy<IOperation>(() =>
+                    new LocalReferenceExpression(local, isDeclaration, _semanticModel, syntax, type, constantValue, isImplicit));
+                return new LazyDeclarationExpression(localReferenceExpression, _semanticModel, declarationExpressionSyntax, type, constantValue: default, isImplicit: false);
+            }
             return new LocalReferenceExpression(local, isDeclaration, _semanticModel, syntax, type, constantValue, isImplicit);
```

Comment:
In each of these cases (`Call`, `FieldAccess`, `TupleExpression`), consider creating the `new Local*ReferenceExpression()` once, before the `if`. Then it looks like a helper method could be extracted to cover all three cases: `return LocalReferenceOrDeclaration(localReferenceExpression, isDeclaration, _semanticModel, syntax, type);`.

Reply:
Ah, never mind, I see the reference is lazy in the declaration expression case.